### PR TITLE
feat(scripts): retrieve blocks script

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- This PR introduces a script to retrieve blocks from the network (paseo, westend, kusama, polkadot)
- This scripts is needed to make our lives less painful in order to debug a specific block on the Westend (or any other network)
- The script enables to retrieve a single block or a chain of blocks, the response is stored in an output file in the hexadecimal form, in the README I also explain how to decode the response into a `network.BlockResponse` struct and start using/debugging the blocks

## Tests

```sh
go test -timeout 10m -run ^TestBuildRequestMessage$ github.com/ChainSafe/gossamer/scripts/retrieve_block --tags=integration -v
```

## Issues

- N/A

## Primary Reviewer

@jimjbrettj

### CODE REVIEW

File: scripts/retrieve_block/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+## retrieve_block
+
+The script is used to retrieve blocks from the network
```
Comment (jimjbrettj) [inline scripts/retrieve_block/README.md:3]: awesome readme to accompany this script!

File: scripts/retrieve_block/retrieve_block.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+	}
+
+	externalAddr, err = ma.NewMultiaddr(fmt.Sprintf("/ip4/%s/tcp/%d", ip, port))
+	if err != nil {
+		log.Fatalf("creating external address: %s", err.Error())
+	}
+
+	// Set your own keypair
+	priv, _, err := crypto.GenerateKeyPair(
+		crypto.Ed25519, // Select your key type. Ed25519 are nice short
```
Comment (jimjbrettj) [inline scripts/retrieve_block/retrieve_block.go]: This is a manual step needed? Should this be highlighted in the readme?

**Review by jimjbrettj — APPROVED** [review]
This is a super useful script, great job being proactive on this and for taking the time to make it!

File: scripts/retrieve_block/retrieve_block.go (hunk 2 above)
Comment (EclesioMeloJunior) [reply scripts/retrieve_block/retrieve_block.go]: Oh, removed the comments, is not needed to do any manual step

File: scripts/retrieve_block/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+> _note that the arguments to request a chain of blocks are separated by comma_
+
+> _the max numbers of blocks you can retrieve is 128. Given network limitations in the implementation the peer might not respond the full chain you've requested_
+
+```sh
+go run retrieve_block.go [number or hash],[direction],[number of blocks] [network chain spec] [output file]
+
+# requesting from a chain of blocks from 10 to 13
+go run retrieve_block.go 10,asc,3 ./westend.json file.out
```
Comment (edwardmack) [inline scripts/retrieve_block/README.md]: for requesting blocks 10 to 12

**Review by edwardmack — APPROVED** [review]
This looks great, it will be very useful for debugging.  One minor correction to the readme.

**Pull request merged**